src/Dependencies/Collections/ImmutableSegmentedHashSet`1+ValueBuilder.cs

```diff
@@ -0,0 +1,282 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.Collections.Internal;
+
+namespace Microsoft.CodeAnalysis.Collections
+{
+    internal readonly partial struct ImmutableSegmentedHashSet<T>
+    {
+        private struct ValueBuilder : ISet<T>, IReadOnlyCollection<T>
+        {
+            /// <summary>
+            /// The immutable collection this builder is based on.
+            /// </summary>
+            private ImmutableSegmentedHashSet<T> _set;
+
+            /// <summary>
+            /// The current mutable collection this builder is operating on. This field is initialized to a copy of
+            /// <see cref="_set"/> the first time a change is made.
+            /// </summary>
+            private SegmentedHashSet<T>? _mutableSet;
+
+            internal ValueBuilder(ImmutableSegmentedHashSet<T> set)
+            {
+                _set = set;
+                _mutableSet = null;
+            }
+
+            /// <inheritdoc cref="ImmutableHashSet{T}.Builder.KeyComparer"/>
+            public IEqualityComparer<T> KeyComparer
+            {
+                get
+                {
+                    return ReadOnlySet.Comparer;
+                }
+
+                set
+                {
+                    if (Equals(KeyComparer, value ?? EqualityComparer<T>.Default))
+                        return;
+
+                    _mutableSet = new SegmentedHashSet<T>(ReadOnlySet, value ?? EqualityComparer<T>.Default);
+                    _set = default;
+                }
+            }
+
+            /// <inheritdoc cref="ImmutableHashSet{T}.Builder.Count"/>
+            public int Count => ReadOnlySet.Count;
+
+            internal SegmentedHashSet<T> ReadOnlySet => _mutableSet ?? _set._set;
+
+            bool ICollection<T>.IsReadOnly => false;
+
+            private SegmentedHashSet<T> GetOrCreateMutableSet()
```

Comment:
>GetOrCreateMutableSet [](http://example.com/codeflow?start=40&length=21)

Why does this impl differ so much from ImmutableSegmentedDictionary.ValueBuilder.GetOrCreateMutableDictionary?

Reply:
➡️ Updated the one in ImmutableSegmentedDictionary to match the one for hash set, which also matches the existing one for list.